documentation/authentication.md

```diff
@@ -0,0 +1,135 @@
+
+If you're running on Windows, you can save these settings to `%USERPROFILE%\.dotnet-monitor\settings.json`. If you're on other operating systems, you can save this file to `$XDG_CONFIG_HOME/dotnet-monitor/settings.json`.
+
+> **NOTE:** If `$XDG_CONFIG_HOME` isn't defined, dotnet-monitor will fall back to looking at `$HOME/.config/dotnet-monitor/settings.json`
+
+```json
+{
+  "ApiAuthentication": {
+    "ApiKeyHash": "<HASHED-TOKEN>",
+    "ApiKeyHashType": "SHA256"
+  }
+}
+```
+
+Alternatively, you can use environment variables to specify the configuration.
+
+```sh
+DotnetMonitor_ApiAuthentication__ApiKeyHash="<HASHED-TOKEN>"
+DotnetMonitor_ApiAuthentication__ApiKeyHashType="SHA256"
+```
+
+> **NOTE:** When you use environment variables to configure the API Key hash, you must restart `dotnet-monitor` for the changes to take effect.
+
+### Configuring an API Token in a Kubernetes Cluster
+If you're running in Kubernetes, we recommend creating secrets and mounting them into container via a deployment manifest.
+
+```sh
+kubectl create secret generic apikey \
+	--from-literal=ApiAuthentication__ApiKeyHash=$hash \
+	--dry-run=client -o yaml \
+	| kubectl apply -f -
+```
+
+Mount the secret as a file into `/etc/dotnet-monitor` in the deployment manifest for your application (abbreviated for clarity).
+
+```yaml
+spec:
+  volumes:
+  - name: apikey
+    secret:
+      secretName: apikey
+  containers:
+  - name: dotnetmonitoragent
+    image: mcr.microsoft.com/dotnet/dotnet-monitor:5.0.0-preview.4
+    volumeMounts:
+      - name: apikey
+        mountPath: /etc/dotnet-monitor
+
+```
+
+> **NOTE:** For a complete example of running dotnet-monitor in Kubernetes, see [Running in a Kubernetes Cluster](getting-started.md#running-in-a-kubernetes-cluster) in the Getting Started guide.
+
+### Generate an API Token with PowerShell
+```powershell
+$rng = New-Object System.Security.Cryptography.RNGCryptoServiceProvider
+$secret = [byte[]]::new(32)
+$rng.GetBytes($secret)
+$API_TOKEN = [Convert]::ToBase64String($secret)
+"Authorization: MonitorApiKey $API_TOKEN"
+
+$secretStream = [System.IO.MemoryStream]::new($secret)
+$HASHED_TOKEN = (Get-FileHash -Algorithm SHA256 -InputStream $secretStream).Hash
+$rng.Dispose()
+
+Write-Output "ApiKeyHash: $HASHED_TOKEN"
+Write-Output "ApiKeyHashType: SHA256"
+```
+
+### Generate an API Token with a Linux shell
+
+```sh
+API_TOKEN=`openssl rand -base64 32`
+echo "Authorization: MonitorApiKey $API_TOKEN"
+
+HASHED_TOKEN=`$API_TOKEN | base64 -d | sha256sum`
+echo "ApiKeyHash: $HASHED_TOKEN"
+echo "ApiKeyHashType: SHA256"
+```
+
+## Authenticating requests
+When using Windows Authentication, your browser with automatically handle the Windows authentication challenge. If you are using an API Token, you must specify it via the authentication header.
+
+```sh
+curl -H "Authorization: MonitorApiKey HdFb8OPkE0Dc5hpu0kAxA7hhNguAah9SNUFftlP2Dk0=" https://localhost:52323/processes
```

Comment:
Could you use the same encoded key throughout the example: `Uf6Yq8ZGkcn+ltq2QLHxuXpKA6/Q5VH5Mb5aSIJBxRc=`